src/components/common/animations/ShareOrCopyButton/ShareOrCopyButtonAnimated.js

```diff
@@ -1,24 +1,37 @@
 import React from 'react'
-import Lottie from 'lottie-react-native'
-import { TouchableOpacity } from 'react-native'
 import { isString } from 'lodash'
-import AnimationBase from '../Base'
-import { isMobileReactNative } from '../../../../lib/utils/platform'
-import { withStyles } from '../../../../lib/styles'
+import Lottie from 'lottie-react-native'
+import { Platform, TouchableOpacity } from 'react-native'
 import useNativeSharing from '../../../../lib/hooks/useNativeSharing'
 import { isSharingAvailable } from '../../../../lib/share'
 import { useClipboardCopy } from '../../../../lib/hooks/useClipboard'
 
-import animationData from './data.json'
+import AnimationBase from '../Base'
+import { isMobileReactNative } from '../../../../lib/utils/platform'
+import { withStyles } from '../../../../lib/styles'
+
+import animationDataCopy from './data_copy.json'
+import animationDataShare from './data_share.json'
 
-class ShareLinkReceiveButton extends AnimationBase {
+class ShareButton extends AnimationBase {
   state = {
     disabled: false,
     performed: false,
   }
 
+  playUntil = 90 //shorten dead animation
```

Comment:
any options to the react components should be passed via props
any default values should be set via defaultProps

Reply:
thats a fixed constant specific to this animation